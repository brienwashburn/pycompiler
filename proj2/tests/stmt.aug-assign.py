a += b
a -= b
a *= b
a /= b
a %= b
a &= b
a |= b
a ^= b
a <<= b
a >>= b
a **= b
a //= b

a += b + c
a -= b * f * rax
a *= b / 10
a /= b + q
a %= b | 1
a &= b ^ foo
a |= b & bar

a += b,
a += b,c
a += b,c,d,e,f,g,
a **= 3,2
