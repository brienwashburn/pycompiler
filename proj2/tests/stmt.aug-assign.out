(Module
 (AugAssign (Name a) Add (Name b))
 (AugAssign (Name a) Sub (Name b))
 (AugAssign (Name a) Mult (Name b))
 (AugAssign (Name a) Div (Name b))
 (AugAssign (Name a) Mod (Name b))
 (AugAssign (Name a) BitAnd (Name b))
 (AugAssign (Name a) BitOr (Name b))
 (AugAssign (Name a) BitXor (Name b))
 (AugAssign (Name a) LShift (Name b))
 (AugAssign (Name a) RShift (Name b))
 (AugAssign (Name a) Pow (Name b))
 (AugAssign (Name a) FloorDiv (Name b))
 (AugAssign (Name a) Add (BinOp (Name b) Add (Name c)))
 (AugAssign
  (Name a)
  Sub
  (BinOp (BinOp (Name b) Mult (Name f)) Mult (Name rax)))
 (AugAssign (Name a) Mult (BinOp (Name b) Div (Num 10)))
 (AugAssign (Name a) Div (BinOp (Name b) Add (Name q)))
 (AugAssign (Name a) Mod (BinOp (Name b) BitOr (Num 1)))
 (AugAssign (Name a) BitAnd (BinOp (Name b) BitXor (Name foo)))
 (AugAssign (Name a) BitOr (BinOp (Name b) BitAnd (Name bar)))
 (AugAssign (Name a) Add (Tuple (Name b)))
 (AugAssign (Name a) Add (Tuple (Name b) (Name c)))
 (AugAssign
  (Name a)
  Add
  (Tuple (Name b) (Name c) (Name d) (Name e) (Name f) (Name g)))
 (AugAssign (Name a) Pow (Tuple (Num 3) (Num 2))))
